packages/react-composites/src/composites/CallWithChatComposite/adapter/AzureCommunicationCallWithChatAdapter.ts

```diff
@@ -177,18 +192,115 @@ export class AzureCommunicationCallWithChatAdapter implements CallWithChatAdapte
     };
 
     this.callAdapter.onStateChange(onCallStateChange);
+    /* @conditional-compile-remove(breakout-rooms) */
+    this.callAdapter.on('breakoutRoomsUpdated', async (eventData: BreakoutRoomsEventData) => {
+      if (eventData.type === 'join') {
+        await this.breakoutRoomJoined(eventData.data);
+      } else if (eventData.type === 'assignedBreakoutRooms') {
+        if (!eventData.data || eventData.data.state === 'closed') {
+          await this.returnFromBreakoutRoom();
+        }
+      }
+    });
+    /* @conditional-compile-remove(breakout-rooms) */
+    this.callAdapter.on('callEnded', () => {
+      const originCallId = this.context.getState().call?.breakoutRooms?.breakoutRoomOriginCallId;
+
+      // If the call ended is a breakout room call, return to the origin call chat adapter
+      if (originCallId && this.originCallChatAdapter) {
+        this.breakoutRoomChatAdapter?.dispose();
+        this.updateChatAdapter(this.originCallChatAdapter);
+      }
+    });
     this.onCallStateChange = onCallStateChange;
   }
 
+  /* @conditional-compile-remove(breakout-rooms) */
+  private async breakoutRoomJoined(call: Call | TeamsCall): Promise<void> {
+    const targetThreadId = call.info.threadId;
+
+    // If the chat adapter is not on the target thread then we need to switch to the breakout room chat adapter
+    if (targetThreadId && this.chatAdapter && this.chatAdapter.getState().thread.threadId !== targetThreadId) {
+      // Unsubscribe from chat adapter state changes
+      this.chatAdapter.offStateChange(this.onChatStateChange);
+      // Set chat state to undefined to prevent showing chat thread of origin call
+      this.context.unsetChatState();
+
+      // Check if the breakout room chat adapter has been initialized
+      this.breakoutRoomChatAdapter = await this.setBreakoutRoomChatAdapterToThread(targetThreadId);
+
+      // Wait for the user to be added to the thread of chat adapter before updating the current chat adapter
+      // to avoid chat errors of not having access to the chat thread. This delayed access to the chat thread
+      // is also seen in Teams
+      // Check up to 20 times every 500ms and then continue.
+      await busyWait(() => {
+        // If the call adapter's call id has been changed, then stop waiting and don't update the chat adapter
+        if (this.context.getState().call?.id !== call.id) {
+          return true;
+        }
+
+        const userAddedToThread =
+          !!this.breakoutRoomChatAdapter?.getState().thread.participants[
```

Comment:
So can we access the `.thread.participants` without having been added to the thread yet?

Reply:
I suppose its just empty array?

Reply:
Yes it is just an empty array